odl/discr/discretization.py

```diff
@@ -450,9 +448,7 @@ class DiscretizationVector(RawDiscretizationVector, FnBaseVector):
 
     def __init__(self, space, data):
         """Initialize a new instance."""
-        if not isinstance(space, Discretization):
-            raise TypeError('space {!r} not a `Discretization` '
-                            'instance.'.format(space))
+        assert isinstance(space, Discretization)
         super().__init__(space, data)
```

Comment:
Perhaps `FnBaseVector.__init__` instead? Performance and lets you see which of the super-classes is called at a glance.